packages/make-styles/src/makeStyles.ts

```diff
@@ -1,53 +1,85 @@
 import { createCSSVariablesProxy } from './runtime/createCSSVariablesProxy';
 import { resolveClassesBySlots } from './runtime/resolveClassesBySlots';
 import { resolveStyleRules } from './runtime/resolveStyleRules';
-import { MakeStylesOptions, MakeStylesStyleFunctionRule, MakeStylesStyleRule, ResolvedStylesBySlots } from './types';
+import {
+  ResolvedClasses,
+  ResolvedCSSRules,
+  MakeStylesOptions,
+  MakeStylesStyleFunctionRule,
+  MakeStylesStyleRule,
+  StyleBucketName,
+} from './types';
+
+export type StylesBySlots<Slots extends string, Tokens> = Record<Slots, MakeStylesStyleRule<Tokens>>;
+
+export function resolveStyles<Slots extends string, Tokens>(
+  stylesBySlots: StylesBySlots<Slots, Tokens>,
+  unstable_cssPriority: number,
+): [ResolvedClasses<Slots>, ResolvedCSSRules] {
+  const tokensProxy = createCSSVariablesProxy() as Tokens;
+
+  const resolvedClassesBySlots = {} as ResolvedClasses<Slots>;
+  const resolvedCSSRules: ResolvedCSSRules = {};
+
+  // eslint-disable-next-line guard-for-in
+  for (const slotName in stylesBySlots) {
+    const slotStyles =
+      typeof stylesBySlots[slotName] === 'function'
+        ? (stylesBySlots[slotName] as MakeStylesStyleFunctionRule<Tokens>)(tokensProxy)
+        : stylesBySlots[slotName];
+
+    const [resolvedStyleRulesForSlot, resolvedCSSRulesForSlot] = resolveStyleRules(slotStyles, unstable_cssPriority);
+
+    resolvedClassesBySlots[slotName] = resolvedStyleRulesForSlot;
+    (Object.keys(resolvedCSSRulesForSlot) as StyleBucketName[]).forEach(styleBucketName => {
+      resolvedCSSRules[styleBucketName] = (resolvedCSSRules[styleBucketName] || []).concat(
+        resolvedCSSRulesForSlot[styleBucketName]!,
```

Comment:
you shouldn't need the exclamation mark, they type you declared doesn't include undefined values

```suggestion
        resolvedCSSRulesForSlot[styleBucketName],
```

Reply:
nvm it's wrapped in `Partial`